packages/edit-post/src/editor.js

```diff
@@ -170,6 +167,8 @@ function Editor( {
 		return hasThemeStyles ? settings.styles : settings.defaultEditorStyles;
 	}, [ settings, hasThemeStyles ] );
 
+	usePreParsePatterns();
```

Comment:
Can we inject this hook into the "RootContainer" of the `BlockList` component maybe to avoid having to add it to all custom block editor implementations?

Reply:
Or I can refactor the hook to use global state, in that case, it doesn't matter how many times it's called

Reply:
> LiveBlockPreview would also trigger the hook in that case. Maybe we can register it as a plugin?

that seems fine we should just make sure there's no patterns passed to the block editor used in the previews.

Reply:
`BlockList` rerenders too often. Basically whenever I move my mouse over a block then `BlockList` rerenders. Thus resetting the parsing index in the `usePreParsePatterns`.

Reply:
> Basically whenever I move my mouse over a block then BlockList rerenders. Thus resetting the parsing index in the usePreParsePatterns.

Doesn't this mean that there's something wrong in the hook, the hook's effect shouldn't retrigger just because the the component using it serenaders. Unless one of the effects dependencies change, it shouldn't retrigger. Maybe we're missing to define deps in useEffect calls?